apps/vs-code-designer/src/app/commands/createNewCodeProject/cloudToLocal.ts

```diff
@@ -0,0 +1,199 @@
+// Used createNewCodeProject.ts as a template to create this file
+// This file is used to take a zipped Logic App from the desktop and unzip to the local workspace
+// Reorganized file with constants at the top, grouped functions, and a main cloudToLocalInternal to call everything
+import {
+  extensionCommand,
+  funcVersionSetting,
+  projectLanguageSetting,
+  projectOpenBehaviorSetting,
+  projectTemplateKeySetting,
+  parameterizeConnectionsInProjectLoadSetting,
+} from '../../../constants';
+import { localize } from '../../../localize';
+import { addLocalFuncTelemetry, tryGetLocalFuncVersion, tryParseFuncVersion } from '../../utils/funcCoreTools/funcVersion';
+import { showPreviewWarning } from '../../utils/taskUtils';
+import { getGlobalSetting, getWorkspaceSetting } from '../../utils/vsCodeConfig/settings';
+import { OpenBehaviorStep } from '../createNewProject/OpenBehaviorStep';
+import { FolderListStep } from '../createNewProject/createProjectSteps/FolderListStep';
+import { NewCodeProjectTypeStep } from './CodeProjectBase/NewCodeProjectTypeStep';
+import { ZipFileStep } from '../createNewProject/createProjectSteps/ZipFileStep';
+import { OpenFolderStepCodeProject } from './CodeProjectBase/OpenFolderStepCodeProject';
+import { SetLogicAppName } from './CodeProjectBase/SetLogicAppNameStep';
+import { setWorkspaceName } from './CodeProjectBase/SetWorkspaceName';
+import { extend } from '@microsoft/logic-apps-shared';
+import { AzureWizard } from '@microsoft/vscode-azext-utils';
+import type { IActionContext } from '@microsoft/vscode-azext-utils';
+import { latestGAVersion, OpenBehavior } from '@microsoft/vscode-extension-logic-apps';
+import { extractConnectionDetails, changeAuthTypeToRaw, updateConnectionKeys, getConnectionsJsonContent } from './cloudToLocalHelper';
+import type { ICreateFunctionOptions, IFunctionWizardContext, ProjectLanguage } from '@microsoft/vscode-extension-logic-apps';
+import { window } from 'vscode';
+import * as path from 'path';
+import AdmZip = require('adm-zip');
+import * as fs from 'fs';
+import { SetLogicAppType } from './CodeProjectBase/setLogicAppType';
+import { writeFormattedJson } from '../../utils/fs';
+
+// Constants
+const openFolder = true;
+
+// Function to create an AdmZip instance
+function createAdmZipInstance(zipFilePath: string) {
+  return new AdmZip(zipFilePath);
+}
+
+// Function to get zip entries
+function getZipEntries(zipFilePath: string) {
+  const zip = createAdmZipInstance(zipFilePath);
+  return zip.getEntries();
+}
+
+async function cleanLocalSettings(localSettingsPath: string) {
+  const localSettings = JSON.parse(fs.readFileSync(localSettingsPath, 'utf8'));
+
+  if (localSettings.Values) {
+    Object.keys(localSettings.Values).forEach((key) => {
```

Comment:
nit: it would be more performant to just change these keys directly if they exist instead of iterating through the entire array. 

ie.  if(localSettings.Values[key].exists  (or whatever the equivalent is in typescript).